src/TrackGenerator.cpp

```diff
@@ -677,6 +684,7 @@ void TrackGenerator::initializeTracks() {
   double* phi_eff = new double[_num_azim];
 
   double x1, x2;
+  double z = _geometry->getZLevel();
```

Comment:
@samuelshaner I was wondering your thoughts - it seems to me that the z level concept is more applicable to the `TrackGenerator` than to the `Geometry`. Would you agree? Could it be easily transferred to the `TrackGenerator` from the `Geometry`?